geom/type_multi_polygon.go

```diff
@@ -91,8 +103,8 @@ func validateMultiPolygon(polys []Polygon, opts ctorOptionSet) error {
 				polyBoundaries[j],
 			)
 			if !interMLS.IsEmpty() {
-				return validationError{"multipolygon child polygon " +
-					"boundaries intersect at multiple points"}
+				return defyChildPolysMultiTouch.errAtPt(
```

Comment:
What does `defy` mean in the context of reporting errors here?

Reply:
`defy` refers to the validation rule being defied. I think there are two issues here, though:

- `defyRingsMultiTouch` and `defyChildPolysMultiTouch` as names are inverted compared to the others (i.e. they describe what's wrong rather than what the rule is).
- `defy` is short, but maybe not the right word to use. `violation` is probably better.

I'll have a think about better ways to name these.

Reply:
Yup, `violation` sounds good. Other alternatives are `break` or `breach`. I thought [this is a nice explanation](https://www.italki.com/en/post/question-231972) of these variants:

> 1) "breach" usually refers to a barrier of some sort, such as a wall. e.g. "Sir, the wall has been breached by the enemy. Should we fall back now??"
> 2) "break" is the most commonly used of the above four words, and can be used in almost any context
> 3) "violate" usually refers to a law or right, such as "that country violates it's peoples' human rights" or a person's integrity ("violate" in that context is often synonymous with rape")
> 4) "defy" usually refers to a regime, or some kind of other person or institution, it usually infers willful action against another person's wishes

Reply:
Cool thanks for that explanation. I think those descriptions confirm that violate is the best word to use. Updated in da510a0b49fd2a9d522148abf06462255c7baccf.